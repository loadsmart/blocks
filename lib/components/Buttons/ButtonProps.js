"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ButtonDisplayState;
(function (ButtonDisplayState) {
    ButtonDisplayState[ButtonDisplayState["Normal"] = 0] = "Normal";
    ButtonDisplayState[ButtonDisplayState["Disabled"] = 1] = "Disabled";
    ButtonDisplayState[ButtonDisplayState["Loading"] = 2] = "Loading";
})(ButtonDisplayState = exports.ButtonDisplayState || (exports.ButtonDisplayState = {}));
var ButtonHeight;
(function (ButtonHeight) {
    ButtonHeight[ButtonHeight["Regular"] = 50] = "Regular";
    ButtonHeight[ButtonHeight["Short"] = 42] = "Short";
})(ButtonHeight = exports.ButtonHeight || (exports.ButtonHeight = {}));
