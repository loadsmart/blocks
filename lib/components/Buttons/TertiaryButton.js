"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importStar(require("react"));
var react_native_1 = require("react-native");
var res_1 = require("../../res/");
var TertiaryButtonDisplayStyle;
(function (TertiaryButtonDisplayStyle) {
    TertiaryButtonDisplayStyle[TertiaryButtonDisplayStyle["Dark"] = 0] = "Dark";
    TertiaryButtonDisplayStyle[TertiaryButtonDisplayStyle["Light"] = 1] = "Light";
})(TertiaryButtonDisplayStyle = exports.TertiaryButtonDisplayStyle || (exports.TertiaryButtonDisplayStyle = {}));
var TertiaryButton = /** @class */ (function (_super) {
    __extends(TertiaryButton, _super);
    function TertiaryButton() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TertiaryButton.prototype.render = function () {
        var styles = stylesFromProps(this.props);
        return (react_1.default.createElement(react_native_1.TouchableOpacity, { onPress: this.props.onPress },
            react_1.default.createElement(react_1.default.Fragment, null,
                react_1.default.createElement(react_native_1.Text, { style: [styles.title, this.props.style] }, this.props.title))));
    };
    return TertiaryButton;
}(react_1.PureComponent));
exports.default = TertiaryButton;
var stylesFromProps = function (props) {
    var isLight = props.displayStyle === TertiaryButtonDisplayStyle.Light;
    return react_native_1.StyleSheet.create({
        title: {
            color: isLight ? res_1.Colors.White : res_1.Colors.Charcoal,
            textDecorationLine: 'underline',
            fontFamily: res_1.Fonts.SharpSansMedium,
            fontSize: 14,
            textAlign: 'center',
        },
    });
};
