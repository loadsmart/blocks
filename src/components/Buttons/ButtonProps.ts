import { ImageSourcePropType, ImageStyle, StyleProp, ViewProps } from 'react-native'

export enum ButtonDisplayState {
  Normal,
  Disabled,
  Loading,
}

export enum ButtonHeight {
  Regular = 50,
  Short = 42,
}

export enum ButtonDisplayStyle {
  Dark,
  Light,
}

export interface ButtonProps extends ViewProps {
  icon?: ImageSourcePropType
  iconStyle?: StyleProp<ImageStyle>
  title: string
  buttonHeight?: ButtonHeight
  displayState?: ButtonDisplayState
  displayStyle?: ButtonDisplayStyle
  onPress?: () => void
}
