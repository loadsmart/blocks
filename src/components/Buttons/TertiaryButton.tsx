import React, { PureComponent } from 'react'
import { StyleSheet, Text, TouchableOpacity, ViewProps } from 'react-native'
import { Colors, Fonts } from '../../res/'

export enum TertiaryButtonDisplayStyle {
  Dark,
  Light,
}

interface Props extends ViewProps {
  title: string
  displayStyle?: TertiaryButtonDisplayStyle
  onPress?: () => void
}

export default class TertiaryButton extends PureComponent<Props> {
  public render() {
    const styles = stylesFromProps(this.props)
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <>
          <Text style={[styles.title, this.props.style]}>{this.props.title}</Text>
        </>
      </TouchableOpacity>
    )
  }
}

const stylesFromProps = (props: Props) => {
  const isLight = props.displayStyle === TertiaryButtonDisplayStyle.Light
  return StyleSheet.create({
    title: {
      color: isLight ? Colors.White : Colors.Charcoal,
      textDecorationLine: 'underline',
      fontFamily: Fonts.SharpSansMedium,
      fontSize: 14,
      textAlign: 'center',
    },
  })
}
