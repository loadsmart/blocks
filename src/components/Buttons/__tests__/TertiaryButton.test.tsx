import React from 'react'
import renderer from 'react-test-renderer'
import TertiaryButton from '../TertiaryButton'
import { Text, TouchableOpacity } from 'react-native'
import { ButtonDisplayStyle } from '../ButtonProps'

describe('TertiaryButton', () => {
  it('renders title', () => {
    const testRenderer = renderer.create(<TertiaryButton title={'Skip'} />)

    const title = testRenderer.root.findByType(Text)
    expect(title.props.children).toBe('Skip')
  })

  it('executes onPress callback', () => {
    const onPress = jest.fn()
    const testRenderer = renderer.create(<TertiaryButton title={'Skip'} onPress={onPress} />)

    const touchable = testRenderer.root.findByType(TouchableOpacity)
    touchable.props.onPress()

    expect(onPress).toHaveBeenCalled()
  })

  it('renders for style dark', () => {
    const testRenderer = renderer.create(
      <TertiaryButton title={'Skip'} displayStyle={ButtonDisplayStyle.Dark} />
    )

    expect(testRenderer.toJSON).toMatchSnapshot()
  })

  it('renders for style light', () => {
    const testRenderer = renderer.create(
      <TertiaryButton title={'Skip'} displayStyle={ButtonDisplayStyle.Light} />
    )

    expect(testRenderer.toJSON).toMatchSnapshot()
  })
})
