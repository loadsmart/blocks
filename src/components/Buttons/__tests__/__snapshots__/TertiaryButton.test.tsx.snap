// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TertiaryButton renders for style dark 1`] = `[Function]`;

exports[`TertiaryButton renders for style light 1`] = `[Function]`;
